database/transaction.rs

```diff
@@ -163,20 +163,15 @@ impl<D: DurabilityClient> TransactionWrite<D> {
         }
     }
 
-    pub fn commit(mut self) -> (TransactionProfile, Result<(), DataCommitError>) {
-        self.profile.commit_profile().start();
-        let (mut profile, result) = self.try_commit();
-        profile.commit_profile().end();
-        (profile, result)
-    }
-
-    pub fn try_commit(self) -> (TransactionProfile, Result<(), DataCommitError>) {
+    pub fn finalise_snapshot(self) -> (TransactionProfile, Result<(DatabaseDropGuard<D>, WriteSnapshot<D>), DataCommitError>) {
```

Comment:
Generate a commitable write snapshot out of this transaction

Reply:
looks good. Could also just be finalise() imo if you wanted to keep it simpler naming